pkg/controller/hostpathprovisioner/daemonset.go

```diff
@@ -100,6 +114,20 @@ func (r *ReconcileHostPathProvisioner) reconcileDaemonSet(reqLogger logr.Logger,
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileHostPathProvisioner) deleteDaemonSet(name, namespace string) error {
+	// Check if this DaemonSet already exists
+	ds := &appsv1.DaemonSet{}
```

Comment:
There is no value in getting and then deleting, something in between might delete and we would not properly handle the error from the delete in that case, you can simply delete like this:
```go
if err := r.client.Delete(context.TODO(), ds); err != nil && !errors.IsNotFound(err) {
  return err
}
return nil
```
Make sure to do this for all the objects.

Reply:
Done

Reply:
And also tidied up the delete error checking as we discussed.